src/UuidFactory.php

```diff
@@ -178,6 +183,29 @@ public function fromInteger($integer)
         return $this->fromString($hex);
     }
 
+    /**
+     * Checks if a uuid string is valid
+     *
+     * @param  string  $uuid The string to check
+     * @return boolean       Returns true if the string represents a valid UUID; otherwise, returns false
+     */
+    public function isValid($uuid)
+    {
+        // assert or InvalidArgumentException should be used for the integrity of the argument
+        // These statements give helpful indication during debugging
+        assert(func_num_args() === 1);
+        assert('is_string($uuid)');
+
+        $uuid = str_replace(array('urn:', 'uuid:', '{', '}'), '', $uuid);
+
+        // This should probably be moved to the UuidFactory from Uuid
+        if ($uuid === Uuid::NIL || preg_match('/' . self::VALID_PATTERN . '/', $uuid)) {
```

Comment:
I prefer to keep the `NIL` constant on the Uuid class. The Uuid class is the base class of the library. All the other namespace constants, etc. live there. Most use-cases involve simply invoking the static methods on the Uuid class to get UUIDs. The factory and interface are provided to allow others a convenient way to replace certain generators with their own implementations, but not many are using these in practice.